backend/src/main/kotlin/org/loculus/backend/model/ReleasedDataModel.kt

```diff
@@ -85,31 +84,35 @@ open class ReleasedDataModel(
-            ("submitter" to TextNode(rawProcessedData.submitter)) +
-            ("groupId" to IntNode(rawProcessedData.groupId)) +
-            ("groupName" to TextNode(rawProcessedData.groupName)) +
-            ("submittedDate" to TextNode(rawProcessedData.submittedAtTimestamp.toUtcDateString())) +
-            ("submittedAtTimestamp" to LongNode(rawProcessedData.submittedAtTimestamp.toTimestamp())) +
-            ("releasedAtTimestamp" to LongNode(rawProcessedData.releasedAtTimestamp.toTimestamp())) +
-            ("releasedDate" to TextNode(rawProcessedData.releasedAtTimestamp.toUtcDateString())) +
-            ("versionStatus" to TextNode(versionStatus.name)) +
-            ("dataUseTerms" to TextNode(currentDataUseTerms.type.name)) +
-            ("dataUseTermsRestrictedUntil" to restrictedDataUseTermsUntil) +
-            ("versionComment" to TextNode(rawProcessedData.versionComment))
-
-        if (backendConfig.dataUseTermsUrls != null) {
-            val url = if (rawProcessedData.dataUseTerms == DataUseTerms.Open) {
-                backendConfig.dataUseTermsUrls.open
-            } else {
-                backendConfig.dataUseTermsUrls.restricted
+            mapOf(
+                ("accession" to TextNode(rawProcessedData.accession)),
+                ("version" to LongNode(rawProcessedData.version)),
+                (HEADER_TO_CONNECT_METADATA_AND_SEQUENCES to TextNode(rawProcessedData.submissionId)),
+                ("accessionVersion" to TextNode(rawProcessedData.displayAccessionVersion())),
+                ("isRevocation" to BooleanNode.valueOf(rawProcessedData.isRevocation)),
+                ("submitter" to TextNode(rawProcessedData.submitter)),
+                ("groupId" to IntNode(rawProcessedData.groupId)),
+                ("groupName" to TextNode(rawProcessedData.groupName)),
+                ("submittedDate" to TextNode(rawProcessedData.submittedAtTimestamp.toUtcDateString())),
+                ("submittedAtTimestamp" to LongNode(rawProcessedData.submittedAtTimestamp.toTimestamp())),
+                ("releasedAtTimestamp" to LongNode(rawProcessedData.releasedAtTimestamp.toTimestamp())),
+                ("releasedDate" to TextNode(rawProcessedData.releasedAtTimestamp.toUtcDateString())),
+                ("versionStatus" to TextNode(versionStatus.name)),
+                ("dataUseTerms" to TextNode(currentDataUseTerms.type.name)),
+                ("dataUseTermsRestrictedUntil" to restrictedDataUseTermsUntil),
+                ("versionComment" to TextNode(rawProcessedData.versionComment)),
+            ).let {
+                when (backendConfig.dataUseTermsUrls) {
+                    null -> it
+                    else -> {
+                        val url = when (currentDataUseTerms) {
+                            DataUseTerms.Open -> backendConfig.dataUseTermsUrls.open
+                            is DataUseTerms.Restricted -> backendConfig.dataUseTermsUrls.restricted
```

Comment:
Very naïve question: Is this correct or do we have to write `is DataUseTerms.Open`?

Reply:
GH copilot chat says:
In the provided code, the `currentDataUseTerms` variable is checked in a `when` clause to determine which URL to use for the `dataUseTermsUrl` field. The selection shows:

```kotlin
val url = when (currentDataUseTerms) {
    DataUseTerms.Open -> backendConfig.dataUseTermsUrls.open
    is DataUseTerms.Restricted -> backendConfig.dataUseTermsUrls.restricted
}
```

The use of `is` for `DataUseTerms.Restricted` and not for `DataUseTerms.Open` is correct and necessary. Here's why:

- `DataUseTerms.Open` is likely a specific object or singleton instance of the `DataUseTerms` type, which means it can be directly compared using the equality operator (`==`), or in this context, it can be used directly in the `when` clause.
- `is DataUseTerms.Restricted` is used because `DataUseTerms.Restricted` is probably a class (or a sealed subclass) that can hold additional data (e.g., a restricted until date). The `is` keyword checks if `currentDataUseTerms` is an instance of the `DataUseTerms.Restricted` class (or subclass). 

Therefore, using `is` for `Restricted` and not for `Open` ensures that the correct type (and any associated data) is handled appropriately.

Reply:
Exactly. `DataUseTerms.Open` is an `object` (and thus a singleton). `Restricted` is a class.